### DISCUSSION TITLE
How to set cookie with JS hooks?

### DISCUSSION THREAD

**Discussion by VictorioBerra** [pocketbase/pocketbase#4564] [Q&A] [answered]
GoLang:

```GoLang
		e.HttpContext.SetCookie(&http.Cookie{
			Name:     cookieName,
			Value:    e.Token,
			Expires:  time.Now().Add(time.Second * time.Duration(sets().RecordAuthToken.Duration)),
			Path:     "/",
			SameSite: http.SameSiteStrictMode,
		})
```

How can I do the same with JS?

`e.httpContext.SetCookie()`

```
TypeError: Object has no member 'SetCookie'
```

**Comment by ganigeorgiev** [comment, marked as answer]
The JSVM method names are `camelCased`. You can try with:

```js
e.httpContext.setCookie(new Cookie({ ... }))
```
If you haven't already, you can reference the builtin ambient TS declarations at the top of your `.pb.js` file to assist you autocomplete and documentation on hover:

```js
/// <reference path="../pb_data/types.d.ts" />
```
(_For HTML version of the above types you can check https://pocketbase.io/jsvm/index.html_)
